Removing the IsActive clause for dropdown

tomspeak opened the pull request:
When using the poll module, I wanted to set up some inactive polls but you cannot add choices to an inactive poll as the drop down is populated by `$polls = DataObject::get('Poll', '"IsActive" = 1');` I don't think the user should be restricted by this.

Thoughts? Did you have a reason for building it like this @mateusz?

Thanks!

dhensby commented:
So this allows someone to create an inactive poll from the start and add options to it... so they can activate it later?

What reason would there be to create a poll in an "inactive" state?

Though this limitation does seem arbitrary and undocumented (so easy to confuse a user). So I think this is a good idea

tomspeak commented:
Currently the flow is:
- Create a poll and select it to be inactive
- Then go to the `Data` tab to add choices
- Try to select the inactive poll from the `Belongs to` drop down
- The inactive poll is not shown

It would be nice to have complete control of the poll's data regardless of its current state.

dhensby commented:
My question was more "why would anyone create a poll and make it instantly inactive" - but I agree it's not an impossible situation.

mateusz commented:
Yeah, seems artificial. The only concern that springs into mind is if there is 5 active polls, and 10,000 archival inactive polls, how to you find these 5 from the dropdown? Maybe at least they could be sorted, the active on top.

tomspeak commented:
Valid points all around. Going to go back and re-think how to implement this. My initial thoughts are:
- Searchable drop down
- A `GroupedDropdownField`

Will try a few ideas, let me know if you think of any in the mean time.

tomspeak commented:
I've had a play around and it turns out you can have a `DropdownField` with built in search functionality if there are >= 23 items to display. See: https://db.tt/tadazwfe

I also changed the sort to be by `Title ASC` as I feel now that we're pulling in all polls regardless of state, this would make finding polls a little bit easier.

How do you feel about this? I asked a question on the forums: http://www.silverstripe.org/customising-the-cms/show/39323 to find out how to show the search when there's less than 23 items displayed. 

I think if we can get it changed to a value we want, this would be a good change.

Also just added a title for the poll option as it looked a bit out of place without a title!

dhensby commented:
That field was created by @harvest called [chosen](https://github.com/harvesthq/chosen)

It's probably defined in the JS for that... (at a guess).

I do think a grouped dropdown would be better

tomspeak commented:
@dhensby It would be great if a `GroupedDropdownField` works in the same way and auto adds the search, then we could have something like:

```
Search
Active:
    Example 1
    Example 2
    Example 3
Inactive:
    Example 4
    Example 5
    Example 6
```

Would be the best of both worlds.

dhensby commented:
Does it not work that way?

tomspeak commented:
I assume so, I will try it tomorrow -- if it does work as described above do we agree it would be the best implementation?

dhensby commented:
For sure

mateusz commented:
The requirement for this functionality is unclear anyway, and everything your proposed sounds like improvement to me :-)

mateusz commented on code/PollChoice.php:
```diff
@@ -29,13 +29,19 @@ class PollChoice extends DataObject {
-		$pollsMap = array();
-		if($polls) $pollsMap = $polls->map('ID', 'Title', '--- Select a poll ---');
-		
+		$activePolls = DataObject::get('Poll', '"IsActive" = 1');
```
How about using the new ORM ?

``` php
DataObject::get('Poll')->filter(array('IsActive'=>'1'));
```

tomspeak replied:
Good catch! Sorry about these commits, they're a bit all over the place, will clean them up tomorrow!

mateusz commented on code/Poll.php:
```diff
@@ -36,7 +36,7 @@ class Poll extends DataObject implements PermissionProvider {
 	); 
 	
-	static $default_sort = 'Created DESC';
+	static $default_sort = 'Title ASC';
```
Also, while we are at it, this should use private static, as per the new config system requirements :-)
